src/linuxX64Main/kotlin/mu/KotlinLoggingConfiguration.kt

```diff
@@ -0,0 +1,9 @@
+package mu
+
+import kotlin.native.concurrent.AtomicReference
+
+object KotlinLoggingConfiguration {
+    val logLevel: AtomicReference<KotlinLoggingLevel> = AtomicReference(KotlinLoggingLevel.INFO)
```

Comment:
Perhaps it would be better to hide atomics as implementation details and expose plain properties with getters and setters:
```
private val _prop = AtomicReference<Type>...
val prop: Type
    get() = _prop.value
    set(value) { return _prop.value = value }
```

Reply:
Although this would be the way to go there are two issues. 

First issue is that Kotlin 1.4 is due to be released soon (likely in a matter of weeks). and it remains unknown if the public/private properties [feature](https://youtrack.jetbrains.com/issue/KT-14663) will be part of the Kotlin 1.4 release. If the feature is part of the release then that would make the approach a wasted effort.

Second issue is that there is some urgency to allow enough time for @oshai to test the changes before the PR is merged in a timely manner. The comment (review) appeared after a agreement was made for the PR to be merged, **after** the PR has passed a final round of tests.

Reply:
As far as I understand this is going to be public API. If so, you won't be able to change it later without breaking binary and backward compatibility. If you make proper API now then later you can just get rid of private "backing fields" here, but the API won't change. But up to you guys, I'm just a random reviewer here.

Reply:
@napperley - you're effort making this PR is well appreciated! I do see a point in what @arkivanov said and I wonder what do you think in term of the API itself?
In regards to that I am going to release a snapshot version with current implementation anyway before merging, and we can decide that after the merge and before the next release to create another PR to change it, thus not hurting backward compatibility. WDYT?

Reply:
Considering the importance of API backward compatibility the extra time needed to incorporate the changes won't be a issue (this addresses the first issue). As for the second issue it might be better to do the changes anyway, and take chances with the Kotlin 1.4 release (the risk would likely be low).

Reply:
All the changes have been made.